### ISSUE TITLE
bug: [vue] If nested route first level has an argument, all routes will render twice.

### ISSUE THREAD

**Issue by asika32764** [ionic-team/ionic-framework#25254]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x
- [ ] Nightly

### Current Behavior

For some reason, we need a multi-level nested routes and needs arguments.

We found that if the 1st level parent route has an argument, then all routes will double rendered in `IonRouterOutlet`

![p-001-2022-05-08-02-13-40](https://user-images.githubusercontent.com/1639206/167267208-1187e916-4201-4f75-89ed-fff2e745acef.jpg)

### Expected Behavior

Should not render twice.

## Without Argument

```js
const routes: Array<RouteRecordRaw> = [
  {
    path: '/',
    redirect: '/tabs/tab1'
  },
  {
    path: '/tabs',
    component: TabsPage,
    children: [
      {
        path: '',
        redirect: '/tabs/tab1'
      },
      {
        path: 'tab1',
        component: () => import('@/views/Tab1Page.vue'),
        children: [
          {
            path: '',
            component: () => import('@/views/Tab1/Tab1Root.vue'),
          },
```

The routes rendered correctly, every route only render once.

![p-001-2022-05-08-02-08-15](https://user-images.githubusercontent.com/1639206/167267042-d175a084-f986-4b09-ba6a-80bf15ca9a9d.jpg)

## With Argument

```js
const routes: Array<RouteRecordRaw> = [
  {
    path: '/',
    redirect: '/tabs/1/tab1'
  },
  {
    path: '/tabs/:tid', // <-- Here Adds argument
    component: TabsPage,
    children: [
      {
        path: '',
        redirect: '/tabs/1/tab1'
      },
      {
        path: 'tab1',
        component: () => import('@/views/Tab1Page.vue'),
        children: [
          {
            path: '',
            component: () => import('@/views/Tab1/Tab1Root.vue'),
          },
```

Then all routes render twice:

![p-001-2022-05-08-02-12-31](https://user-images.githubusercontent.com/1639206/167267088-8cdea9e4-f42a-48f4-9e87-ef8ffbafb205.jpg)

The root route will render 2 same root components in `IonRouterOutlet`

![p-001-2022-05-08-02-13-40](https://user-images.githubusercontent.com/1639206/167267155-648733f2-f811-44f1-b2c8-e13463158998.jpg)

NOTE: We have 3 levels nested routes with arguments, but only first level with argument will cause this issue.

--------------

## Reproduce

- Correct
    - GitHub: https://github.com/asika32764/ionic-nested-route-test/tree/main
    - CodeSandBox: https://codesandbox.io/s/github/asika32764/ionic-nested-route-test/tree/main

- Not correct
    - GitHub: https://github.com/asika32764/ionic-nested-route-test/tree/not-work
    - CodeSandBox: https://codesandbox.io/s/github/asika32764/ionic-nested-route-test/tree/not-work

In CodeSandBox, open Console, you will see the child component mounted logs. Please click any links.

Correct: only 1 log

![p-001-2022-05-08-02-33-33](https://user-images.githubusercontent.com/1639206/167267308-51bea5ab-3911-48f4-a628-4ba7c86dc1f3.jpg)

Not Correct: multiple logs

![p-001-2022-05-08-02-34-13](https://user-images.githubusercontent.com/1639206/167267336-5afed92d-1f05-445b-b2be-04429ce63a97.jpg)

### Code Reproduction URL

https://github.com/asika32764/ionic-nested-route-test/tree/not-work

### Ionic Info

Ionic:

   Ionic CLI       : 6.18.2 (C:\Users\Simon\AppData\Roaming\nvm\v14.18.2\node_modules\@ionic\cli)
   Ionic Framework : @ionic/vue 6.1.4

Capacitor:

   Capacitor CLI      : 3.5.1
   @capacitor/android : not installed
   @capacitor/core    : 3.5.1
   @capacitor/ios     : not installed

Utility:

   cordova-res : not installed globally
   native-run  : 1.5.0

System:

   NodeJS : v14.18.2 (C:\Program Files\nodejs\node.exe)
   npm    : 6.14.15
   OS     : Windows 10

**Comment by asika32764** [comment]
Tab is just for example to reproduce this issue. The real usage is that we need a projects -> works -> jobs -> tasks scope to handle complex project managements.

See GIF

![routes](https://user-images.githubusercontent.com/1639206/167460195-0313c684-f60e-4db4-914c-736f2a81b225.gif)

![routes2](https://user-images.githubusercontent.com/1639206/167460391-b369ebbc-541d-4eea-8681-8e84c7d2ba6f.gif)

**Comment by liamdebeasi** [comment]
Does your real usage still use tabs? If a parent route has a parameter, I would expect that all child routes would need to re-render since the nested routes are dependent on the parameter that is passed into the URL.

**Comment by asika32764** [comment]
No, the real usage not use tabs but use a nested routes with multiple `IonRouterOutlet`

Inside the deep, every components renders twice.

![routes3](https://user-images.githubusercontent.com/1639206/167461527-a1dbe41d-8390-4418-8991-78d5b07978b1.gif)

------

We use nested routers because every routes needs parameters, we must use hooks to get it.

![p-001-2022-05-10-01-11-06](https://user-images.githubusercontent.com/1639206/167461981-6388b169-b7d6-4e28-af52-86eeab362962.jpg)

Use shared URLs and hook every routes is break the DRY principle.

------

So is this expected behavior for Ionic router? I think it should work since vue-router provides this way to organize routes.

**Comment by asika32764** [comment]
Another case, if we go into nested route from another route, this works,. The root components will not render twice and everything works correctly. So I think this should not be a normal situation.

![routes4](https://user-images.githubusercontent.com/1639206/167464874-cc01e4cb-464c-4444-a211-6a534b51ff4d.gif)

**Comment by liamdebeasi** [comment]
Yes this is expected behavior for Ionic. Nested outlets in native mobile apps behave differently than nested outlets in web apps. This may be where the confusion is coming from. Since Ionic is a toolkit for building mobile apps, we try to match the native mobile app behavior.

In mobile apps, each nested outlet is its own navigation stack. In web apps, this is not the case.

Ionic also differs from other app toolkits as we keep multiple pages in the DOM. This allows us to provide advanced gestures such as swipe to go back. As a result, you may see some additional page re-renders when the URL changes.

----

In general, nested outlets should only ever be used when you want to render content in multiple outlets at the same time. The [Shared URLs vs Nested Routes Guide](https://ionicframework.com/docs/vue/navigation#shared-urls-versus-nested-routes) goes over this in a bit more detail.

Even the Vue Router documentation makes the point on nested URLs: https://router.vuejs.org/guide/essentials/nested-routes.html. The important thing to note in the Vue Router example is they are rendering content in multiples outlets at the same time. This is the intended use case for nested routes in a non-Ionic app with Vue Router as well as an Ionic app with Vue Router.

----

So to summarize:

1. Ionic's nested outlets behave a bit differently than nested outlets you may have seen in other app toolkits. This results in a few additional re-renders when the parent route has changed.
2. The application you provided should not be using nested outlets to begin with as it does not meet the intended use case. The same goes for a non-Ionic app with Vue Router. However, the issue is likely more noticeable in Ionic because of the previous point.

Does that help clarify things at all?

**Comment by asika32764** [comment]
Yes I see. I just curious some points

1. Sometimes it works, see https://github.com/ionic-team/ionic-framework/issues/25254#issuecomment-1121379854
2. And this behavior is not exists in Ionic5 as my memory.
3. If some App has 100up or more routes, shared URLs is the only 1 way to implement nested structure? This is not follows the DRY principle. Or we can have a solution for this usage in the future?

**Comment by liamdebeasi** [comment]
Regardless of the issue resolution, this is probably a sign that we (Ionic) need to improve our documentation around routing to better explain the differences between nested and shared URLs.
